workbench-core/datasets/src/ddbDataSetMetadataPlugin.ts

```diff
@@ -240,6 +282,40 @@ export class DdbDataSetMetadataPlugin implements DataSetMetadataPlugin {
     });
   }
 
+  // Returns the current number of datasets for the storage location
+  private async _storeStorageLocationToDdb(
+    storageLocation: StorageLocation,
+    increment: 'increment' | 'decrement'
+  ): Promise<number> {
+    const validatedStorageLocationMetadata = StorageLocationMetadataParser.safeParse({
+      ...storageLocation,
+      resourceType: 'datasetStorageLocation'
+    });
+
+    if (!validatedStorageLocationMetadata.success) {
+      throw new DataSetInvalidParameterError(validatedStorageLocationMetadata.error.message);
+    }
+
+    const storageLocationKey = {
+      pk: this._dataSetKeyType,
+      sk: buildDynamoDbKey(storageLocation.name, this._storageLocationKeyType)
+    };
+
+    const { Attributes } = await this._aws.helpers.ddb.updateExecuteAndFormat({
+      key: storageLocationKey,
+      params: {
+        item: validatedStorageLocationMetadata.data,
+        set: '#datasetCount = if_not_exists(#datasetCount, :zero) + :counter',
+        names: { '#datasetCount': 'datasetCount' },
+        values: { ':counter': increment === 'increment' ? 1 : -1, ':zero': 0 }
```

Comment:
in theory, you can end up with a dataset count of  < 0. Use a condition express to ensure no mitigate impact or have the condition commented above moved here to check for when it is < 1

Reply:
moved the "if count < 1, then delete" condition to this method.